extension/entitystore/extension.go

```diff
@@ -0,0 +1,257 @@
+	return e.mode
+}
+
+func (e *EntityStore) KubernetesMode() string {
+	return e.kubernetesMode
+}
+
+func (e *EntityStore) EKSInfo() *eksInfo {
+	return e.eksInfo
+}
+
+func (e *EntityStore) EC2Info() EC2Info {
+	return e.ec2Info
+}
+
+func (e *EntityStore) SetNativeCredential(client client.ConfigProvider) {
+	e.nativeCredential = client
+}
+
+func (e *EntityStore) NativeCredentialExists() bool {
+	return e.nativeCredential != nil
+}
+
+// CreateLogFileEntity creates the entity for log events that are being uploaded from a log file in the environment.
+func (e *EntityStore) CreateLogFileEntity(logFileGlob LogFileGlob, logGroupName LogGroupName) *cloudwatchlogs.Entity {
+	serviceAttr := e.serviceprovider.logFileServiceAttribute(logFileGlob, logGroupName)
+
+	keyAttributes := e.createServiceKeyAttributes(serviceAttr)
+	attributeMap := e.createAttributeMap()
+	addNonEmptyToMap(attributeMap, ServiceNameSourceKey, serviceAttr.ServiceNameSource)
+
+	return &cloudwatchlogs.Entity{
+		KeyAttributes: keyAttributes,
+		Attributes:    attributeMap,
+	}
+}
+
+// GetMetricServiceNameAndSource gets the service name source for service metrics if not customer provided
+func (e *EntityStore) GetMetricServiceNameAndSource() (string, string) {
+	return e.serviceprovider.getServiceNameAndSource()
+}
+
+// GetServiceMetricAttributesMap creates the attribute map for service metrics. This will be expanded upon in a later PR'S,
+// but for now is just covering the EC2 attributes for service metrics.
+func (e *EntityStore) GetServiceMetricAttributesMap() map[string]*string {
+	return e.createAttributeMap()
+}
+
+// AddServiceAttrEntryForLogFile adds an entry to the entity store for the provided file glob -> (serviceName, environmentName) key-value pair
+func (e *EntityStore) AddServiceAttrEntryForLogFile(fileGlob LogFileGlob, serviceName string, environmentName string) {
+	if e.serviceprovider != nil {
+		e.serviceprovider.addEntryForLogFile(fileGlob, ServiceAttribute{
+			ServiceName:       serviceName,
+			ServiceNameSource: ServiceNameSourceUserConfiguration,
+			Environment:       environmentName,
+		})
+	}
+}
+
+// AddServiceAttrEntryForLogGroup adds an entry to the entity store for the provided log group nme -> (serviceName, environmentName) key-value pair
+func (e *EntityStore) AddServiceAttrEntryForLogGroup(logGroupName LogGroupName, serviceName string, environmentName string) {
+	e.serviceprovider.addEntryForLogGroup(logGroupName, ServiceAttribute{
+		ServiceName:       serviceName,
+		ServiceNameSource: ServiceNameSourceInstrumentation,
+		Environment:       environmentName,
+	})
+}
+
+func (e *EntityStore) AddPodServiceEnvironmentMapping(podName string, serviceName string, environmentName string, serviceNameSource string) {
+	if e.eksInfo != nil {
+		e.eksInfo.AddPodServiceEnvironmentMapping(podName, serviceName, environmentName, serviceNameSource)
+	}
+}
+
+func (e *EntityStore) StartPodToServiceEnvironmentMappingTtlCache(done chan struct{}) {
```

Comment:
nit: Why do we pass in the done channel if it's already part of the `EntityStore`?

Reply:
Makes sense. Add a new commit to remove passing this value

Reply:
Blocking. Being addressed in new commit.